src/services/symbolDisplay.ts

```diff
@@ -26,6 +26,15 @@ namespace ts.SymbolDisplay {
     }
 
     function getSymbolKindOfConstructorPropertyMethodAccessorFunctionOrVar(typeChecker: TypeChecker, symbol: Symbol, location: Node): ScriptElementKind {
+        const roots = typeChecker.getRootSymbols(symbol);
+        // If this is a method from a mapped type, leave as a method so long as it still has a call signature.
+        if (roots.length === 1
+            && first(roots).flags & SymbolFlags.Method
+            // Ensure the mapped version is still a method, as opposed to `{ [K in keyof I]: number }`.
+            && typeChecker.getTypeOfSymbolAtLocation(symbol, location).getNonNullableType().getCallSignatures().length !== 0) {
+            return ScriptElementKind.memberFunctionElement;

```

Comment:
might want to handel get/set accessors the same way as well.

Reply:
It doesn't look like being a get/set accessor affects symbol display normally anyway -- at `class C { get x() { return 0 }; } new C().x` I just see `(property) C.x: number`. It might be useful to see `(get) C.x: number` instead -- do you think we should add that?

Reply:
it does have a different `ScriptElementKind` kind.. so the IDE could show it using a different icon. just checked in VS and we are not treating them differently, we use the same icon for getters, setters and properties. so might not be that important after all.